hack/generator/pkg/astmodel/testdata/SetStringFromOptionalString-Direct.golden

```diff
@@ -27,7 +27,8 @@ func (person Person) ConvertFrom(source *vHub.Person) error {
 func (person Person) ConvertTo(destination *vHub.Person) error {
 
 	// name
-	destination.name = &person.name
+	name := person.name
+	destination.name = &name
```

Comment:
If the value of `person.name` is "", should this explicitly set `destination.name` to nil?

Reply:
I'm not sure that gains us anything, we'll always have a mismatch because the non-optional field can't handle all the values. We'll end up with one of these two scenarios:

| Scenario | `*string` | `string` | `*string` |
| -- | -- | -- | -- | 
| A | **nil** | `""` | `""` |
| A | `""` | `""` | `""` |
| B | **nil** | `""` | **nil** |
| B | `""` | `""` | **nil** |

Either `""` always goes to **nil** (and we lose any original `""`) or it always goes to `""`" (and we lose any original **nil**)

Reply:
Yeah, that makes sense - it's inherently lossy essentially.